### DISCUSSION TITLE
Finding out the breakpoint was triggered

### DISCUSSION THREAD

**Discussion by emad-elsaid** [go-delve/delve#3066] [Q&A] [answered]
I'm trying to create a UI for Delve and I faced a problem when I tried to find which breakpoint was triggered.

So when the application trip a breakpoint it'll pause. the `continue` call in the `Target` will return. Now I need to find out if this was because the program exited? or is it a breakpoint that tripped. 

If it's a breakpoint, which one? and also which Go routine triggered it, finding the stackframe from there is probably easy as it's the one at the top.

Also same problem when I step, next, stepout. I don't know what did I step into or out to. 

Is there like a signals channel I can listen to to give me events out of the target in a meaningful way? 

I'm using the Debugger service, proc package APIs so far.

**Comment by aarzilli** [comment, marked as answer]
The DebuggerState object returned by Command has a list of threads: threads that have triggered a breakpoint will have a breakpoint object in them, they also have a GoroutineID which is `> 0` if they are running a goroutine.
